src/core/net/tcp6.cpp

```diff
@@ -395,4 +834,235 @@ void Tcp::ProcessTimers()
 } // namespace Ip6
 } // namespace ot
 
+/*
+ * Implement TCPlp system stubs declared in tcplp.h.
+ *
+ * Because these functions have C linkage, it is important that only one
+ * definition is given for each function name, regardless of the namespace it
+ * in. For example, if we give two definitions of tcplp_sys_new_message, we
+ * will get errors, even if they are in different namespaces. To avoid
+ * confusion, I've put these functions outside of any namespace.
+ */
+
+using namespace ot;
+using namespace ot::Ip6;
+
+extern "C" {
+
+otMessage *tcplp_sys_new_message(otInstance *aInstance)
+{
+    Instance &instance = AsCoreType(aInstance);
+    Message * message  = instance.Get<ot::Ip6::Ip6>().NewMessage(0);
+
+    if (message)
+    {
+        message->SetLinkSecurityEnabled(true);
+    }
+
+    return message;
+}
+
+void tcplp_sys_free_message(otInstance *aInstance, otMessage *aMessage)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    Message &message = AsCoreType(aMessage);
+    message.Free();
+}
+
+void tcplp_sys_send_message(otInstance *aInstance, otMessage *aMessage, otMessageInfo *aMessageInfo)
+{
+    Instance &   instance = AsCoreType(aInstance);
+    Message &    message  = AsCoreType(aMessage);
+    MessageInfo &info     = AsCoreType(aMessageInfo);
+
+    otLogDebgTcp("Sending TCP segment: payload_size = %d", static_cast<int>(message.GetLength()));
+
+    IgnoreError(instance.Get<ot::Ip6::Ip6>().SendDatagram(message, info, kProtoTcp));
+}
+
+uint32_t tcplp_sys_get_ticks(void)
+{
+    return TimerMilli::GetNow().GetValue();
+}
+
+uint32_t tcplp_sys_get_millis(void)
+{
+    return TimerMilli::GetNow().GetValue();
+}
+
+void tcplp_sys_set_timer(struct tcpcb *aTcb, uint8_t aTimerFlag, uint32_t aDelay)
+{
+    Tcp::Endpoint &endpoint = Tcp::Endpoint::FromTcb(*aTcb);
+    endpoint.SetTimer(aTimerFlag, aDelay);
+}
+
+void tcplp_sys_stop_timer(struct tcpcb *aTcb, uint8_t aTimerFlag)
+{
+    Tcp::Endpoint &endpoint = Tcp::Endpoint::FromTcb(*aTcb);
+    endpoint.CancelTimer(aTimerFlag);
+}
+
+struct tcpcb *tcplp_sys_accept_ready(struct tcpcb_listen *aTcbListen, struct in6_addr *aAddr, uint16_t aPort)
+{
+    Tcp::Listener &listener = Tcp::Listener::FromTcbListen(*aTcbListen);
+    Tcp &          tcp      = listener.GetInstance().Get<Tcp>();
+    struct tcpcb * rv       = (struct tcpcb *)-1;
+
+    if (listener.mAcceptReadyCallback != nullptr)
```

Comment:
nit: use `VerifyOrExit()` or `ExitNow()` to reduce indent below?

Reply:
Done; implemented in the latest update.